File: LOO 2°Parcial/Tarea 5/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Poliformismo_Shape_csharp
{
    public abstract class Shape
    {
        public abstract string shapename();
        public abstract double calcularea();
        public abstract double calcularperimetro ();
        public abstract double calcularvolumen();
    }
}

File: LOO 2°Parcial/Tarea 5/Shape.py
# -*- coding: utf-8 -*-
"""
Created on Sat Apr 10 14:40:04 2021

@author: usuario
"""

from abc import ABC, abstractmethod

class Shape(ABC):
    @abstractmethod
    def shapename(self):
        pass
    
    @abstractmethod
    def calcularea(self):
        pass

    @abstractmethod
    def calcularperimetro(self):
        pass

    @abstractmethod
    def calcularvolumen(self):
        pass